bika/lims/upgrade/v3_2_0_1709.py

```diff
@@ -0,0 +1,109 @@
+    # create sample catalog indexes
+    for idx in sample_indexes:
+        upgrade_utils.addIndex(CATALOG_SAMPLE_LISTING, idx, sample_indexes[idx])
+    # create sample catalog columns
+    for col in sample_columns:
+        upgrade_utils.addColumn(CATALOG_SAMPLE_LISTING, col)
+
+    # define objects to be catalogued
+    at.setCatalogsByType('Sample', [CATALOG_SAMPLE_LISTING, ])
+    logger.info('Recovering samples to reindex')
+    # retrieve bika catalog and portal catalog so that samples can be uncatalogued
+    bika_catalog = getToolByName(portal, 'bika_catalog')
+    portal_catalog = getToolByName(portal, 'portal_catalog')
+    # get brains of objects to be catalogued from bika catalog
+    sample_brains = bika_catalog(portal_type='Sample')
+    i = 0  # already indexed objects counter
+    # reindex the found objects in the new sample catalog and uncatalog them from bika catalog
+    logger.info('Reindexing samples')
+    for brain in sample_brains:
+        if i % 100 == 0:
+            logger.info('Reindexed {}/{} samples'.format(i, len(sample_brains)))
+            if i % 100 == 0:
+                transaction.commit()
+                logger.info('{0} items processed.'.format(i))
+        sample_obj = brain.getObject()
+        sample_obj.reindexObject()
+        # uncatalog samples from bika catalog and portal catalog
+        path_uid = '/'.join(sample_obj.getPhysicalPath())
+        bika_catalog.uncatalog_object(path_uid)
+        portal_catalog.uncatalog_object(path_uid)
+        i += 1
+    transaction.commit()
+    logger.info('Reindexed {}/{} samples'.format(len(sample_brains), len(sample_brains)))
+
+def update_bika_catalog(portal, upgrade_utils):
+    # remove unnecessary columns and indexes from bika catalog
+    # now that samples have their own catalog
+    logger.info('Updating bika catalog')
+    bika_catalog = getToolByName(portal, 'bika_catalog')
+    unnecessary_cols = ['getSampleID', 'getRequestID', 'getClientOrderNumber', 'getClientReference',
+                        'getClientSampleID', 'getSamplePointTitle', 'getSampleTypeTitle',
+                        'getDatePublished', ]
+    unnecessary_idxs = ['getDepartmentUIDs', 'getAnalysesUIDs', 'getClientOrderNumber', 'getClientReference',
+                        'getClientSampleID', 'getClientTitle', 'getClientUID', 'getDatePublished',
+                        'getInvoiced', 'getRequestID', 'getSampleID', 'getSamplePointTitle',
+                        'getSamplePointUID', 'getSampler', 'getScheduledSamplingSampler',
+                        'getSampleTypeTitle', 'getSampleTypeUID', 'getSampleUID', 'getSamplingDate',
+                        'getWorksheetTemplateTitle', ]
+    for column in unnecessary_cols:
+        upgrade_utils.delColumn(bika_catalog, column)
+    logger.info('Removed unnecessary columns')
+    for index in unnecessary_idxs:
+        upgrade_utils.delIndex(bika_catalog, index)
+    logger.info('Removed unnecessary indexes')
```

Comment:
@juangallostra , @Espurna  . Since this PR will be merged into the wip branch, that currently is v1.0.0, you need to create a new upgrade step (`v01_01_000.py`) instead of using this old one (`v3_2_0_1709.py`). Since the current version of master is `v1.0.0`, note in this case (because it will be merged into wip), minor number increases in 1 unit, while patch remains 0. If this had to be merged into master, then the upgrade step would be `v01_00_001.py` (patch +1)

Reply:
`v01_01_00.py` upgrade step is already there, just add your modifs there